### ISSUE TITLE
[Bug] 3.27.0-beta.1 breaks ember-macro-helpers

### ISSUE THREAD

**Issue by lukemelia** [emberjs/ember.js#19476]
### 🐞 Describe the Bug
The implementation of computed.macro deprecation in 3.27.0-beta.1 breaks [the monkey patch in ember-macro-helpers](https://github.com/kellyselden/ember-macro-helpers/blob/master/vendor/monkey-patch-ember-computed.js#L27).

### 🔬 Minimal Reproduction
This can be reproduced by running the ember-beta scenario of ember-macro-helper's ember-try config.

### 😕 Actual Behavior
In an app with ember-macro-helpers, trying to use, for example: `computed.empty` results in an error because the empty property is not undefined.

### 🤔 Expected Behavior
computed macros such as computed.empty should work normally and emit a deprecation warning.

### 🌍 Environment

- Ember: 3.27.0-beta.1
- Node.js/npm: v12
- OS: Mac
- Browser: Chrome

**Comment by lukemelia** [comment]
I *think* the issue is that the macro properties of `computed` are not enumerable in [the deprecated versions of the properties implementation](https://github.com/emberjs/ember.js/commit/bf750e7f0862bc42b25040773be7ed6cdc47ccce#diff-5a69a5df44575cca36edca671ed7aed75906e802fdbed6feb3547e14a9175aa4R61-R78).

**Comment by rwjblue** [comment]
Ya we can make them enumerable and configurable (which is what it should take to fix). Though ember-macro-helpers will definitely need a re-think to avoid the deprecations.

**Comment by pzuraq** [comment]
Also worth noting that even if we fix enumerability and configurability, direct imports of `computed` macros (e.g. `computed.and`, `computed.on`, etc) will no longer receive the monkey patched version. You would have to migrate to using the nested value directly:

```js
import { computed } from '@ember/object';

computed.and();
```

I think it may be possible for `ember-macro-helpers` to try to patch the module that is loaded in the loader itself, but that is definitely tricky. I've done it before, when Ember Data moved to using real modules for instance.

**Comment by rwjblue** [comment]
The conceptual thing that is done over in `ember-macro-helpers` is just going to have to be rethought (monkey patching Ember's internals).

**Comment by pzuraq** [comment]
Yes, I completely agree, patching the loader IMO would not be a long term fix and `ember-macro-helpers` should immediately deprecate using that itself. I was mainly pointing that out as a potential option to prevent breaking in a minor release.

**Comment by sandstrom** [comment]
I'm doing some issue gardening 🌱🌿 🌷 and came upon this issue. Since it's quite old I just wanted to ask if this is still relevant? If it isn't, maybe we can close this issue?

By closing some old issues we reduce the list of open issues to a more manageable set.

**Comment by kategengler** [comment]
Closing since this has to have been long-since resolved or made irrelevant.
